src/main/java/com/jm/online_store/service/impl/UserServiceImpl.java

```diff
@@ -165,4 +174,51 @@ public boolean activateUser(String token, HttpServletRequest request) {
                 .map(Optional::get)
                 .collect(Collectors.toSet());
     }
+
+    @Override
+    @Transactional
+    public void updateUserImage(Long userId, MultipartFile file) {
+        String originalFilename = StringUtils.cleanPath(file.getOriginalFilename());
+        if (!file.isEmpty()) {
+            deleteUserImage(userId);
+            Path fileNameAndPath = Paths.get(uploadDirectory, originalFilename);
+            try {
+                byte[] bytes = file.getBytes();
+                Files.write(fileNameAndPath, bytes);
+                //Set user's profile picture
+                User user = userRepository.findById(userId).get();
+                user.setProfilePicture(originalFilename);
+                userRepository.save(user);
+            } catch (IOException e) {
+                log.debug("Failed to store file {} {}", fileNameAndPath, e.getMessage());
+            }
+        }
+        log.debug("Failed to store file - file is not present {}", originalFilename);
+    }
+
+    @Override
+    @Transactional
+    public void deleteUserImage(Long userId) {
+        final String defaultAvatar = StringUtils.cleanPath("def.jpg");
+        User user = userRepository.findById(userId).get();
+        //Get profilePicture name from User and delete this profile picture from Uploads
+        Path fileNameAndPath = Paths.get(uploadDirectory, user.getProfilePicture());
+        //Check if deleting picture is not a default avatar
+        try {
+            if (!fileNameAndPath.getFileName().toString().equals(defaultAvatar)) {
+                Files.delete(fileNameAndPath);
+            }
+        } catch (IOException e) {
+            log.debug("Failed to delete file {} {} ", fileNameAndPath.getFileName().toString(), e.getMessage());
```

Comment:
log.debug("Failed to delete file: {}, because: {} ", fileNameAndPath.getFileName().toString(), e.getMessage());